plugins/browser-plugin-media-tracking/src/findMediaElement.ts

```diff
@@ -0,0 +1,38 @@
+export function findMediaElem(mediaId: string): HTMLAudioElement | HTMLVideoElement | null {
+  let el: HTMLVideoElement | HTMLAudioElement | HTMLElement | null = document.getElementById(mediaId);
+
+  if (el instanceof HTMLVideoElement) {
+    // Plyr loads in an initial blank video with currentSrc as https://cdn.plyr.io/static/blank.mp4
+    // so we need to check until currentSrc updates
+    if (el.src === 'https://cdn.plyr.io/static/blank.mp4' || !el.src) {
+      return null;
+    }
+    return el as HTMLVideoElement;
+  }
+
+  if (el && el.tagName !== 'AUDIO' && el.tagName !== 'VIDEO') {
+    el = findMediaElementChild(el);
+  }
+
+  if (el) {
+    if (el.tagName === 'AUDIO') return el as HTMLAudioElement;
+    if (el.tagName === 'VIDEO') return el as HTMLVideoElement;
+  }
+
+  return null;
+}
+
+function findMediaElementChild(el: Element): HTMLAudioElement | HTMLVideoElement | null {
+  let tags: string[] = ['AUDIO', 'VIDEO'];
+
+  for (let tag of tags) {
+    let elem = el.getElementsByTagName(tag);
+    if (elem.length === 1) {
+      if (tag === 'AUDIO') return elem[0] as HTMLAudioElement;
+      if (tag === 'VIDEO') return elem[0] as HTMLVideoElement;
+    } else if (elem.length > 1) {
+      console.error(`There is more than one child ${tag.toLowerCase()} element in the provided node.`);
+    }
+  }
+  return null;
+}
```

Comment:
I'd like to suggest a couple of changes here. First, reuse of strings inflates the filesize, so making AUDIO and VIDEO constants will reduce that (the variable name will get minified to something like `a5` which is much shorter than repeating `AUDIO`). I've also removed the console.error, for two reasons, strings are expensive from a filesize point of view, and logging errors this deep on the code makes them tricky to live with.

I've also prefered const over let where possible (they all become `var` but it makes the developer experience nicer if we're consistant with our expectations on how a variable should be used).

I'd suggest hoisting up the error message into where you call `findMediaElem()` in `index.ts`. You can also then use the core Logger functionality (demonstrated here: https://github.com/snowplow/snowplow-javascript-tracker/blob/268f6cef26edb87b51fda39126fd7ffbc41dff5b/plugins/browser-plugin-consent/src/index.ts#L97).

```suggestion
const videoTag = 'VIDEO', audioTag = 'AUDIO';

export function findMediaElem(mediaId: string): HTMLAudioElement | HTMLVideoElement | null {
  const el: HTMLElement | null = document.getElementById(mediaId);

  if (!el) return null;
  if (isAudioElement(el)) return el;
  if (isVideoElement(el)) {
    // Plyr loads in an initial blank video with currentSrc as https://cdn.plyr.io/static/blank.mp4
    // so we need to check until currentSrc updates
    if (el.src === 'https://cdn.plyr.io/static/blank.mp4' || !el.src) {
      return null;
    }
    return el;
  }

  return findMediaElementChild(el);
}

function findMediaElementChild(el: Element): HTMLAudioElement | HTMLVideoElement | null {
  const tags: string[] = [audioTag, videoTag];

  for (let tag of tags) {
    const elem = el.getElementsByTagName(tag);
    if (elem.length === 1) {
      if (isAudioElement(elem[0])) return elem[0];
      if (isVideoElement(elem[0])) return elem[0];
    }
  }
  return null;
}

function isAudioElement(el: Element): el is HTMLAudioElement {
  return el.tagName === audioTag ? true : false;
} 

function isVideoElement(el: Element): el is HTMLVideoElement {
  return el.tagName === videoTag ? true : false;
} 
```